x-pack/plugins/security_solution/server/endpoint/ingest_integration.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { factory as policyConfigFactory } from '../../common/endpoint/models/policy_config';
+import { NewPolicyData } from '../../common/endpoint/types';
+import { NewDatasource } from '../../../ingest_manager/common/types/models';
+
+/**
+ * Callback to handle creation of Datasources in Ingest Manager
+ * @param newDatasource
+ */
+export const handleDatasourceCreate = async (
```

Comment:
For the most part, this is just a placeholder handler. I currently only moved over the logic that inserts the policy into the Datasource from the Front end Middleware

Reply:
This is great for purposes of this PR! We'll ultimately need to get some things from `CoreStart` and `StartPlugins` in here, but that can be done with a closure or using the app context. Thanks!